index_bidding_history1.php

```diff
@@ -0,0 +1,134 @@
+        </li>
+        <li class="nav-item">
+          <a class="nav-link" href="/onlineauction/logout.php">logout</a>
+        </li>
+      </ul>
+    </div>
+  </div>
+</nav>
+
+
+    <div class="container-fuild">
+    
+        <table class="table">
+            <thead class="table-dark">
+              <tr><h1 style="text-align:center">Bidding history lists</h1></tr>
+                <tr>
+                    <th style="width:5%"> Bid ID</th>
+                    <th style="width:10%"> Product Name</th>
+                    <th> Product ID</th>
+                    <th style="width:15%"> Picture</th>
+                    <th> Bidding Status</th>
+                    <th> Your price</th>
+                    <th> Current Highest price</th>
+                    <th> Bidding Result</th>
+                    <th> Bidding End Date</th>
+                </tr>
+            </thead>
+            <tbody>
+              <?php
+                 $mail = $_SESSION['email'] ;
+                 $sql = "SELECT * FROM bidding JOIN item ON bidding.item_ID = item.item_ID JOIN users ON users.user_ID = bidding.buyer_ID WHERE users.user_type IN ('buyer', 'both') AND users.email = '$mail' order by item.item_id, bidding.bid_id";  # select data from sql table
+                 
+                 //$sql_maxPrice = "SELECT Max(bid_price) AS max_price, item_ID FROM bidding GROUP BY item_ID";
+
+
+                 // echo $sql;
+                 $result = mysqli_query($db_conn, $sql); # send a query to the database
+                 //$result_maxPrice = mysqli_query($db_conn, $sql_maxPrice);
+                 
+
+                 $resultCheck = mysqli_num_rows($result); # check if you can get the data from the databas
+                 //$resultCheck_maxPrice = mysqli_num_rows($result_maxPrice); 
+                 
+                // #fetch the data into an array
+                if ($resultCheck > 0) 
+                {
+                    while ($row = mysqli_fetch_assoc($result)) {
+
+                     $max_price = 0;
+                     $sql_maxPrice = "SELECT item.item_ID, max(bidding.bid_price) AS max_price FROM bidding JOIN item ON bidding.item_ID = item.item_ID JOIN users ON users.user_ID = bidding.buyer_ID WHERE users.user_type IN ('buyer', 'both') AND users.email != '$mail' group by item.item_ID order by item.item_id";  # select data from sql table 
+                     //"SELECT Max(bid_price) AS max_price, item_ID FROM bidding GROUP BY item_ID";
+                    //  AND users.email = '$mail'
+                     $result_maxPrice = mysqli_query($db_conn, $sql_maxPrice);
+                     $resultCheck_maxPrice = mysqli_num_rows($result_maxPrice); 
+                      if($resultCheck_maxPrice > 0) {
+                        while ($row2 = mysqli_fetch_assoc($result_maxPrice))
+                        { 
+                          if ($row['item_ID'] === $row2['item_ID'] AND  bidStatus($row) === "Finished") {
```

Comment:
You can keep only  `bidStatus($row) === "Finished"` in your condition after removing this loop.